mautic/mautic
# TPROD-143 Replaces Froala with CKEditor 4 as the default WYSIWYG

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features" 
| Bug fix?                               | no
| New feature?                           | yes
| Deprecations?                          | yes
| BC breaks?                             | no
| Automated tests included?              | yes
| Related user documentation PR URL      | n/a
| Related developer documentation PR URL | n/a
| Issue(s) addressed                     |  n/a

#### Description:

We've been locked into a very old version of Froala with no upgrade path due to Froala changing their licensing years ago. Thus, this PR replaces Froala with CKEditor 4 everywhere except for the the legacy builder since that is being replaced with Grapesjs. Froala's code will be removed once the legacy builder's code is removed. 

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Create new Dynamic Web Content. CKEditor should be loaded and usable. 
3. Similarly use the different areas in the product that require a WYSIWYG editor (descriptions, etc).
3. Use the legacy builder to build an email or landing page which should still be usable with Froala. 

#### Deprecations
Froala is on its way out so nothing new should be dependent on it.
